Okta/OktaExternalUserAgentIOS.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Foundation
+import OktaAppAuth
+import SafariServices
+
+class OktaExternalUserAgentIOS : NSObject, OIDExternalUserAgent {
+    
+    private let presentingViewController: UIViewController
+    fileprivate var safariViewController: SFSafariViewController?
+    
+    fileprivate var session: OIDExternalUserAgentSession?
+    
+    fileprivate var externalUserAgentFlowInProgress: Bool = false
+    
+    init(presenting presentingViewController: UIViewController) {
+        self.presentingViewController = presentingViewController
+    }
+
+    func present(_ request: OIDExternalUserAgentRequest, session: OIDExternalUserAgentSession) -> Bool {
+        guard !externalUserAgentFlowInProgress else {
+            return false
+        }
+        
+        guard let requestURL = request.externalUserAgentRequestURL() else {
+            return false
+        }
+        self.session = session
+        externalUserAgentFlowInProgress = true
+        var isBrowserLaunched = false
+        
+        if #available(iOS 9.0, *) {
```

Comment:
This seems like duplicated code to me. Inside of [OIDExternalUserAgent](https://github.com/openid/AppAuth-iOS/blob/0c822671e56cedf159a6d27ccf0669b689c0f2b5/Source/iOS/OIDExternalUserAgentIOS.m), they detect which iOS version is being used. Why do we need to copy this logic?

Reply:
Also - before this gets merged, I'd like to see testing around this new flow.

Ideally we will have tests that assert:
1) On iOS 9, `SFAuthenticationSession ` is the browser used to remove the session
2) On iOS 11+, the **native** browser is used to remove the session
3) If a session was **created** with `SFAuthenticationSession`, it can be removed with the **native** browser.

Reply:
@jmelberg-okta I'd like to clarify that SFAuthenticationSession won't be used for Okta logout. We will use SFSafariViewController for iOS 9+, and redirect to a native browser for older OSs.